src/perf/lib/PerfServer.h

```diff
@@ -15,6 +15,72 @@
 #include "SecNetPerf.h"
 #include "Tcp.h"
 
+class PerfServer;
+
+struct StreamContext {
+    StreamContext(
+        PerfServer* Server, bool Unidirectional, bool BufferedIo) :
+        Server{ Server }, Unidirectional{ Unidirectional }, BufferedIo{ BufferedIo } {
+        if (BufferedIo) {
+            IdealSendBuffer = 1; // Hack to get just do 1 send at a time.
+        }
+    }
+    CXPLAT_HASHTABLE_ENTRY Entry; // To TCP StreamTable
+    PerfServer* Server;
+    const bool Unidirectional;
+    const bool BufferedIo;
+    bool ResponseSizeSet{ false };
+    bool SendShutdown{ false };
+    bool RecvShutdown{ false };
+    uint64_t IdealSendBuffer{ PERF_DEFAULT_SEND_BUFFER_SIZE };
+    uint64_t ResponseSize{ 0 };
+    uint64_t BytesSent{ 0 };
+    uint64_t OutstandingBytes{ 0 };
+    QUIC_BUFFER LastBuffer;
+};
+
+typedef enum SYNTHETIC_DELAY_TYPE {
+    SYNTHETIC_DELAY_FIXED,
+    SYNTHETIC_DELAY_VARIABLE
+} SYNTHETIC_DELAY_TYPE;
+
+
+struct DelayedWorkContext {
+    StreamContext* Context;
+    void* Handle;
+    bool IsTcp;
+    DelayedWorkContext* Next;
+};
+
+class DelayWorker {
+public:
+    PerfServer* m_Server = nullptr;
+    CXPLAT_EXECUTION_CONTEXT ExecutionContext = { 0 };
+    bool Initialized{ false };
+    CXPLAT_THREAD Thread;
+    CXPLAT_EVENT WakeEvent;
+    CXPLAT_EVENT DoneEvent;
+    CXPLAT_DISPATCH_LOCK Lock;
```

Comment:
I don't _think_ you need a dispatch lock. MsQuic only ever indicates events up to the app at PASSIVE. So you should be able to use `CxPlatLock`.